nighthawk/test/benchmark_http_client_test.cc

```diff
@@ -0,0 +1,336 @@
+#include <chrono>
+
+#include "ares.h"
+
+#include "gtest/gtest.h"
+
+#include "common/api/api_impl.h"
+#include "common/common/compiler_requirements.h"
+#include "common/common/thread_impl.h"
+#include "common/event/dispatcher_impl.h"
+#include "common/filesystem/filesystem_impl.h"
+#include "common/http/header_map_impl.h"
+#include "common/network/dns_impl.h"
+#include "common/network/utility.h"
+#include "common/runtime/runtime_impl.h"
+#include "common/stats/isolated_store_impl.h"
+#include "common/thread_local/thread_local_impl.h"
+
+#include "nighthawk/source/client/benchmark_client_impl.h"
+#include "nighthawk/source/common/platform_util_impl.h"
+#include "nighthawk/source/common/rate_limiter_impl.h"
+#include "nighthawk/source/common/sequencer_impl.h"
+#include "nighthawk/source/common/statistic_impl.h"
+
+#include "envoy/thread_local/thread_local.h"
+#include "test/mocks/runtime/mocks.h"
+#include "test/mocks/thread_local/mocks.h"
+
+#include "test/integration/http_integration.h"
+
+#include "test/integration/integration.h"
+#include "test/integration/utility.h"
+#include "test/server/utility.h"
+#include "test/test_common/utility.h"
+
+using namespace std::chrono_literals;
+
+namespace Nighthawk {
+
+class BenchmarkClientTest : public Envoy::BaseIntegrationTest,
+                            public testing::TestWithParam<Envoy::Network::Address::IpVersion> {
+public:
+  BenchmarkClientTest()
+      : Envoy::BaseIntegrationTest(GetParam(), realTime(), BenchmarkClientTest::envoy_config),
+        api_(thread_factory_, store_, timeSystem()), dispatcher_(api_.allocateDispatcher()) {}
+
+  static void copyFileToWorkingDir(const std::string path, const std::string path_to) {
+    Envoy::Filesystem::InstanceImpl filesystem;
+    const std::string content =
+        filesystem.fileReadToEnd(Envoy::TestEnvironment::runfilesPath(path));
+    Envoy::TestEnvironment::writeStringToFileForTest(path_to, content);
+  }
+
+  static void SetUpTestCase() {
+    ASSERT_NE("", Envoy::TestEnvironment::getCheckedEnvVar("TEST_TMPDIR"));
```

Comment:
I think you can assume this from Bazel, since it sets this variable, https://docs.bazel.build/versions/master/test-encyclopedia.html